Add coordinator rotation test

akildemir opened the pull request.

Commit:
add node side unit test

Commit:
complete rotation test for all networks

Commit:
update to develop latest

Commit:
set up the fast-epoch docker file

Commit:
fix pr comments

Commit:
Merge branch 'develop' of https://github.com/serai-dex/serai into vs-rotation-tests

Commit:
add coordinator side rotation test

Commit:
bug fixes

kayabaNerve committed:
Merge branch 'develop' into HEAD

kayabaNerve commented on tests/coordinator/src/tests/key_gen.rs:
```diff
@@ -23,37 +23,48 @@ use crate::tests::*;
+          found = true;
+        }
+        CoordinatorMessage::Substrate(
+          messages::substrate::CoordinatorMessage::ConfirmKeyPair { .. },
```
Why is this block here? I'm unclear why we're not deterministically handling ConfirmKeyPair and instead are ad-hoc dropping them.

kayabaNerve committed:
Remove EPOCH_INTERVAL

kayabaNerve committed:
Minor nits

kayabaNerve committed:
Add note on origin of publish_tx function in tests/coordinator

kayabaNerve committed:
Correct ThresholdParams assert_eq

kayabaNerve committed:
fmt

kayabaNerve committed:
Correct detection of handover completion

kayabaNerve committed:
Restore key gen message match from develop

It was modified in response to the handover completion bug, which has now been
resolved.

Commit:
bug fixes

kayabaNerve committed:
Merge branch 'develop' into HEAD

kayabaNerve committed:
Correct invalid constant

kayabaNerve committed:
Typo fixes

Commit:
remove selecting participant to remove at random
